src/index.ts

```diff
@@ -58,57 +59,89 @@ export function getToolCommand(): string {
-  let output = '';
-  const stdout = (data: Buffer): void => {
-    output += data.toString();
-  };
-  const stderr = (data: Buffer): void => {
-    output += data.toString();
-  };
-  const options = {
-    listeners: {
-      stdout,
-      stderr,
-    },
-    silent,
+export async function gcloudRun(cmd: string[], options?: ExecOptions): Promise<ExecOutput> {
+  const toolCommand = getToolCommand();
+  const opts = Object.assign({}, { silent: true, ignoreReturnCode: true }, options);
+  const commandString = `${toolCommand} ${cmd.join(' ')}`;
+  core.debug(`Running command: ${commandString}`);
+
+  const result = await getExecOutput(toolCommand, cmd, opts);
+  if (result.exitCode !== 0) {
+    const errMsg = result.stderr || `command exited ${result.exitCode}, but stderr had no output`;
+    throw new Error(`failed to execute command \`${commandString}\`: ${errMsg}`);
+  }
+
+  return {
+    stderr: result.stderr,
+    stdout: result.stdout,
+    output: result.stdout + '\n' + result.stderr,
   };
+}
 
-  const toolCommand = getToolCommand();
+/**
+ * gcloudRunJSON runs the gcloud command with JSON output and parses the result
+ * as JSON. If the parsing fails, it throws an error.
+ *
+ * @param cmd The command to run.
+ * @param options Any options.
+ *
+ * @return Parsed JSON as an object (or array).
+ */
+export async function gcloudRunJSON(cmd: string[], options?: ExecOptions): Promise<any> {
+  const jsonCmd = ['--format', 'json'].concat(cmd);
+  const output = await gcloudRun(jsonCmd, options);
 
-  await exec.exec(toolCommand, ['config', 'get-value', 'project'], options);
-  return !output.includes('unset');
+  try {
+    const parsed = JSON.parse(output.stdout);
+    return parsed;
+  } catch (err) {
+    throw new Error(
+      `failed to parse output as JSON: ${err}\n\nstdout:\n${output.stdout}\n\nstderr:\n${output.stderr}`,
+    );
+  }
+}
+
+/**
+ * Checks if the project Id is set in the gcloud config.
+ *
+ * @returns true is project Id is set.
+ */
+export async function isProjectIdSet(): Promise<boolean> {
+  const result = await gcloudRun(['config', 'get-value', 'project']);
+  return !result.output.includes('unset');
 }
 
 /**
  * Checks if gcloud is authenticated.
  *
  * @returns true is gcloud is authenticated.
  */
-export async function isAuthenticated(silent = true): Promise<boolean> {
-  let output = '';
-  const stdout = (data: Buffer): void => {
-    output += data.toString();
-  };
-  const stderr = (data: Buffer): void => {
-    output += data.toString();
-  };
-  const options = {
-    listeners: {
-      stdout,
-      stderr,
-    },
-    silent,
-  };
-
-  const toolCommand = getToolCommand();
-
-  await exec.exec(toolCommand, ['auth', 'list'], options);
-  return !output.includes('No credentialed accounts.');
+export async function isAuthenticated(): Promise<boolean> {
+  const result = await gcloudRun(['auth', 'list']);
+  return !result.output.includes('No credentialed accounts.');
```

Comment:
I saw you used `gcloudRunJson` in the tests for `auth list`. As a followup, we could switch it here too for stability.

Reply:
Yea - it's on my list. I didn't want to change _too_ many things at once :)

Reply:
https://github.com/google-github-actions/setup-cloud-sdk/issues/80